### PR TITLE
Registered model table

### PR DESCRIPTION
**Pull request by dpanshug**
Closes: [RHOAIENG-2231](https://issues.redhat.com/browse/RHOAIENG-2231) [RHOAIENG-2232](https://issues.redhat.com/browse/RHOAIENG-2232)

## Description
Created Registered model table page

<details><summary>Registered model page</summary>

image
</details>

<details><summary>Model registry selector with favorite selection</summary>

Screenshot from 2024-04-05 17-11-40
</details>

<details><summary>Table searching with keyword</summary>

image
</details>

<details><summary>Register new version disable for now</summary>

Screenshot from 2024-04-05 17-12-12
</details>

<details><summary>View archived models disable for now</summary>

Screenshot from 2024-04-05 17-12-10
</details>

<details><summary>Label popover when less than 8</summary>

Screenshot from 2024-04-05 17-12-26
</details>

<details><summary>Label modal when greater than or equal to 8</summary>

image
</details>

<details><summary>Label modal search</summary>

image
</details>

## How Has This Been Tested?
Open Model registry section from sidebar

## Test Impact

Added unit test for `getRegisteredModelLabels`

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

@yih-wang

### CODE REVIEW

**Comment by dpanshug** [comment]
Currently WIP, need some tests and review

File: frontend/src/pages/modelRegistry/screens/RegisteredModelListView.tsx [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+                    variant: 'action',
+                    items: [
+                      <MenuToggleAction
+                        id="register-model-button"
+                        key="register-model-button"
+                        data-testid="register-model-button"
+                        aria-label="Registered model"
+                        onClick={() => undefined}
+                      >
+                        Registered model
```
Comment (manaswinidas) [inline frontend/src/pages/modelRegistry/screens/RegisteredModelListView.tsx:104]: As per the mocks, I think this is `Register model`.

**Review by manaswinidas — CHANGES_REQUESTED** [review]

**Comment by yih-wang** [comment]
@dpanshug Thanks for posting the screenshots! Several comments:
#### Registered model page
image
- The padding between the name and description looks larger than other tables in the current console (e.g. pipeline table, projects table). Does it follow the standard layout?
- Row 4 (fraud detection model) shows the case that labels exceed the regular two rows. Could we also incorporate the '2 more' label into the label groups? This way, all labels including those displaying 'x more', won't exceed two rows, ensuring consistent row heights throughout the UI.

#### Table searching with keyword
image
- Chip row is missing
- I don't know the search logic being applied here (it's probably already implemented as such) so just a reminder that when searching in the model list, we'll also query the version level. In other words, models with versions containing the search keyword in their name, description, or labels will also be included in the search results.

#### Label modal when greater than or equal to 8
image
- We won't truncate any labels in this modal

**Comment by dpanshug** [comment]
@yih-wang, added your suggested changes, thanks

> when searching in the model list, we'll also query the version level.

This will be done in future PRs when we add model version table.

**Comment by yih-wang** [comment]
One more comment: the primary button has a typo. Should be "Register model"

**Review by lucferbux — APPROVED** [review]
/approve

There are a few things that we need to get back really soon but we can continue

**Comment by lucferbux** [comment]
/override ci/prow/images ci/prow/odh-dashboard-pr-image-mirror

**Pull request merged**